Editor/Manifest/Version.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace StansAssets.Foundation.Editor
+{
+    public class Version
+    {
+        // Regular Expression to check a Semantic Version string. With numbered capture groups
+        // (so cg1 = major, cg2 = minor, cg3 = patch, cg4 = prerelease and cg5 = buildmetadata)
+        // that is compatible with ECMA Script (JavaScript),
+        // PCRE (Perl Compatible Regular Expressions, i.e. Perl, PHP and R), Python and Go.
+        // See: https://regex101.com/r/vkijKf/1/
+        static readonly Regex s_Matcher = new Regex(@"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$");
+
+        public readonly int Major;
+        public readonly int Minor;
+        public readonly int Patch;
+
+        public readonly bool HasPreRelease;
+        public readonly string PreRelease;
+
+        public readonly bool HasBuildMetadata;
+        public readonly string BuildMetadata;
+
+        string m_StringRepresentation;
+
+        Version(int major, int minor, int patch, string preRelease, string buildMetadata)
```

Comment:
Not like the goal here to make Codacy happy.
But why are we keeping this `priavte` ?

I can see a value for a user to be able to make a version object.
This is not a change requests Just a question.